app/controllers/assets/link_previews/tracks_controller.rb

```diff
@@ -0,0 +1,147 @@
+class Assets::LinkPreviews::TracksController < ApplicationController
+  def show
+    track = Track.find(params[:id])
+
+    map = MiniMagick::Image.open(get_map_url(track))
+    map.contrast
+
+    convert = MiniMagick::Tool::Convert.new
+    convert.size '380x300'
+    convert.xc 'black'
+    convert.draw 'fill #ffffff roundRectangle 0,0 380,300 5,5'
+    convert << 'app/assets/images/canvas.png'
+    convert.call
+
+    card = MiniMagick::Image.open('app/assets/images/canvas.png')
+
+    logo = MiniMagick::Image.open('app/assets/images/logo260.png')
+    logo.resize '24x24'
+
+    result = card.composite(map) do |c|
+      c.compose 'Over'    # OverCompositeOp
+      c.geometry '+10+10' # copy second_image onto first_image from (20, 20)
+    end
+
+    result = result.composite(logo) do |c|
+      c.compose 'Over'    # OverCompositeOp
+      c.geometry '+334+216' # copy second_image onto first_image from (20, 20)
+    end
+
+    result.draw "text 10,270 'Скорость'"
+    result.draw "text 10,285 '375 км/ч'"
+
+    result.draw "text 80,270 'Расстояние'"
+    result.draw "text 80,285 '5975м'"
+
+    result.draw "text 170,270 'Время'"
+    result.draw "text 170,285 '143 с'"
+
+    send_file result.path, type: result.mime_type, disposition: 'inline'
```

Comment:
Two things here:
1. I would extract everything above `send_file` to model that will not inherit from `ApplicationRecord`, just pure class.
2. You missed `authorize` call. That way it will be possible to view data from private tracks.